test/test_client.lua

```diff
@@ -0,0 +1,88 @@
+local nakama = require "nakama.nakama"
+local test_engine = require "nakama.engine.test"
+
+
+context("Nakama client", function()
+
+	before(function()
+		test_engine.reset()
+	end)
+	after(function() end)
+
+	local function config()
+		return {
+			host = "127.0.0.1",
+			port = 7350,
+			use_ssl = false,
+			username = "defaultkey",
+			password = "",
+			engine = test_engine,
+			timeout = 10, -- connection timeout in seconds
+		}
+	end
+
+	local function pprint(t)
+		for k,v in pairs(t) do
+			print(k, v)
+		end
+	end
+
+	test("It should be able to create a client", function()
+		local config = config()
+		local client = nakama.create_client(config)
+		assert_not_nil(client)
+	end)
+
+	test("It should be able to call REST API functions", function()
```

Comment:
The naming of this test isn't accurate. Since we're entirely using mocks here the only things we're verifying are that the `client.authenticate_email` function passes appropriate arguments to the `http` function and that it accurately parses the `response` to create a valid `session` object.

I would split this into two tests. One for validating that `client.authenticate_email` calls the `http` function with the correct parameters, named `"client.authenticate_email sends appropriate HTTP request"`.

Then a second called `"client.authenticate_email returns a valid session object"`  that tests the structure of the returned `session` object is correct.

Or at the very least I would rename the whole test to `"client.authenticate_email sends appropriate HTTP request and returns valid session object"`, as this test doesn't verify that all REST API functions work as intended.